rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java

```diff
@@ -221,6 +222,12 @@ public void didChange(DidChangeTextDocumentParams params) {
         logger.trace("Change contents: {}", params.getTextDocument());
         updateContents(params.getTextDocument(), last(params.getContentChanges()).getText());
         invalidateFacts(params.getTextDocument());
+        ILanguageContributions contrib = contributions(params.getTextDocument());
+        contrib.hasAnalyze().thenAccept(b -> { // TODO: `b` can be cached?
```

Comment:
well, the contribs can be updated (see the `registerContribution`). so we would have to update it every time. where do we normally check this condition? seems a bit strange to do this in this function, I think it should move to `ParametericFileFacts` ?